### PR TITLE
Dashboards/demo styling

### PR DESCRIPTION
**Pull request by nasin**
Style suggestions for dashboard demos

### CODE REVIEW

**Review by bre1470 — CHANGES_REQUESTED** [review]
Looks awesome! One thing in minimal demo: The RDA label of the column chart is better placed on the right side, otherwise the contrast is not strong enough.

File: samples/dashboards/demo/datacursor-sync/demo.js
Comment (sebastianbochan) [inline samples/dashboards/demo/datacursor-sync/demo.js:53]: To remove

File: samples/dashboards/demo/datacursor-sync/demo.js
Comment (sebastianbochan) [inline samples/dashboards/demo/datacursor-sync/demo.js:68]: Replace `});` with `}, false)` to avoid double redraw.

File: samples/dashboards/demo/sync-extremes/demo.css
Comment (sebastianbochan) [inline samples/dashboards/demo/sync-extremes/demo.css:11]: To remove

**Review by sebastianbochan — CHANGES_REQUESTED** [review]

**Review by pawellysy — COMMENTED** [review]
Looks good to me, just a minor comment -> 
There is a bunch of changes in this PR https://github.com/highcharts/highcharts/pull/19306
which are waiting for Torstein's approval, and there is a bit of overlap on the changes.
Should we merge the changes here, or just leave other PR until it gets reviewed?

@pawelfus

**Comment by pawelfus** [comment]
Let's include Torstein in this PR too @pawelfus

File: samples/dashboards/demo/sync-extremes/demo.css
Comment (nasin) [reply samples/dashboards/demo/sync-extremes/demo.css]: done

File: samples/dashboards/demo/datacursor-sync/demo.js
Comment (nasin) [reply samples/dashboards/demo/datacursor-sync/demo.js:67]: done

**Review by bre1470 — APPROVED** [review]
Thanks! Looks good. 👍

File: samples/dashboards/demo/sync-extremes/demo.js
Comment (pawelfus) [inline samples/dashboards/demo/sync-extremes/demo.js:258]: We need to remove all `series` options - it should be possible to get the same effect without `series`. 

Something like this: https://jsfiddle.net/BlackLabel/9zd4aLs1/3/ (in JS tab: modified config for the component, in CSS disabled inactive state at the bottom)

**Review by pawelfus — CHANGES_REQUESTED** [review]
Just one tiny change, other than that it looks good!

File: samples/dashboards/demo/datagrid-sync/demo.js
Comment (sebastianbochan) [inline samples/dashboards/demo/datagrid-sync/demo.js:20]: You don't need to use counter. You can remove the whole load event and use spread operator in component chartOptions.

See the comment below.

File: samples/dashboards/demo/datagrid-sync/demo.js
Comment (sebastianbochan) [inline samples/dashboards/demo/datagrid-sync/demo.js:99]:
```
chartOptions: {
    ...chartOptions,
   {
        chart: {
           events: {
               load: function () {
                   // call update
               }
           }
        }
   }
}
```

**Review by sebastianbochan — CHANGES_REQUESTED** [review]

**Review by pawellysy — APPROVED** [review]

**Review by sebastianbochan — APPROVED** [review]
Looks great!

**Review by pawelfus — APPROVED** [review]
Thanks!

File: samples/dashboards/demo/minimal/demo.css
Comment (TorsteinHonsi) [inline samples/dashboards/demo/minimal/demo.css:1]: This demo has several cases of overlapping and clipped text on small screen widths. Particularly in mobile view, where the three columns in the top probably should be resolved to 100% width.

File: samples/dashboards/demo/minimal/demo.js
Comment (TorsteinHonsi) [inline samples/dashboards/demo/minimal/demo.js:1]: Instead of a separate cell for the title, we should consider a good old `h1` or `h2` here. The frame around the title looks out of place, besides the title text is clipped on small screens.

File: samples/dashboards/demo/minimal/demo.js
Comment (TorsteinHonsi) [inline samples/dashboards/demo/minimal/demo.js:100]: A z index of 20 makes the tooltip render below the plot line. Which looks buggy.

File: samples/dashboards/demo/sync-extremes/demo.js
Comment (TorsteinHonsi) [inline samples/dashboards/demo/sync-extremes/demo.js:1]: It took me a while to make the connection that all the columns represented different data of the same city. Big color legends are not very (cognitively) accessible, so it is hard to read and compare the charts. Instead of doing this split, can we try to make the font size smaller for the city names, and maybe force a vertical rotation?

Comment (TorsteinHonsi) [inline samples/dashboards/demo/sync-extremes/demo.js:1]: This demo too lacks reponsive rules. On mobile screens, the three charts in the top are too narrow. If possible, we should set the breakpoint so high that the axis labels are always readable.

Comment (TorsteinHonsi) [reply samples/dashboards/demo/sync-extremes/demo.js:1]: PS: I may be that those charts would be better off as bar charts (with vertical drag-zoom). Then the labels would always be readable, and as a bonus each city would always be on the same line.

File: samples/dashboards/demo/datacursor-sync/demo.js
Comment (oysteinmoseng) [inline samples/dashboards/demo/datacursor-sync/demo.js:14]: Responsive options (going from small to large) don't seem to take effect unless the page is reloaded, is this a bug or config?

File: samples/dashboards/demo/minimal/demo.css
Comment (oysteinmoseng) [inline samples/dashboards/demo/minimal/demo.css:50]: The iron value color is very low contrast against the background in dark mode. We should probably not try to use the same text colors between light/dark, as that is hard to make work.

File: samples/dashboards/demo/minimal/demo.css
Comment (oysteinmoseng) [inline samples/dashboards/demo/minimal/demo.css:71]: This one is low contrast too in dark mode.

File: samples/dashboards/demo/minimal/demo.js
Comment (oysteinmoseng) [inline samples/dashboards/demo/minimal/demo.js:24]: Is the data grid a bit tall for the content?

Screen Shot 2023-08-19 at 15 19 43

**Review by oysteinmoseng — CHANGES_REQUESTED** [review]
Nice work! Looks a lot cleaner. Added some comments, but I'm not sure how many are relevant to the styling changes, so if any should be separate tasks instead feel free to just add them in Asana.

File: samples/dashboards/demo/datagrid-sync/demo.js
Comment (pawellysy) [reply samples/dashboards/demo/datagrid-sync/demo.js:19]: IMO centered title/subtitle looks good. Maybe we should change the text itself?

File: samples/dashboards/demo/axes-sync/demo.js
Comment (pawellysy) [reply samples/dashboards/demo/axes-sync/demo.js:89]: I guess this is a bug of HTMLComponent.

File: samples/dashboards/demo/datacursor-sync/demo.js
Comment (pawellysy) [reply samples/dashboards/demo/datacursor-sync/demo.js:14]: I think this is a bug

File: samples/dashboards/demo/minimal/demo.js
Comment (pawellysy) [reply samples/dashboards/demo/minimal/demo.js:24]: that's something that happens to the DataGridComponent by default. Something that we might want to add to backlog.

**Review by oysteinmoseng — CHANGES_REQUESTED** [review]
Nice improvements 👍  I will add the relevant non-styling issues to the backlog. 

Couple more things:
- Subtitle in axes-sync demo needs some bottom padding (or remove the background color for the component)
- Climate demo still struggles on mobile (see pic)
Screen Shot 2023-08-22 at 12 22 17

**Review by TorsteinHonsi — APPROVED** [review]
Approved my myself. Now pending @oysteinmoseng's approval.

**Pull request merged**
